### ISSUE TITLE
Noxplayer Android Version 12

### ISSUE THREAD

**Issue by hifiappspro** [HuskyDG/magisk-files#119]
Hello, 
Nox player 12 version I installed magisk delta successfully but after restarting the emulator,
magisk delta installed status shows NA (Not Installed).

**Comment by hifiappspro** [comment]
After I click on Direct Install as System - 
![Screenshot_20230613-133126](https://github.com/HuskyDG/magisk-files/assets/88382598/8aa8ec6a-67bc-4d3f-86d8-a51628abb1fd)
But after i reboot NoxPlayer (Root Option Keep On) i see Magisk not activate after reboot - 
![Screenshot_20230613-134051](https://github.com/HuskyDG/magisk-files/assets/88382598/ec33cf9a-31c7-49b1-a991-3d23cc493513)

**Comment by hifiappspro** [comment]
The emulator has been rebooted and after sending the file can I send the file before reboot?

**Comment by hifiappspro** [comment]
This Folder is not available in System Folder and also not available anywhere in File Manager. Can you already test Magisk Delta on the Android 12 version of Nox Player?

**Comment by VisionR1** [comment]
> Maybe sure you don't disable root access in Nox Player

Yeah, in nox android 12 must have enabled the root options always, with Magisk-fdac22ba-delta(25210), the riru 25.4.4 and zygisk not work anymore like the previous nox android 9

**Comment by maximbc** [comment]
I would like to offer $300 for a quick fix for this.
Husky I tried to contact you on patreon also. Please let me know if there is a way to fix this issue and install magisk delta on nox android 12. Thanks

**Comment by VisionR1** [comment]
I found a magisk.rc in new version 7.0.5.8 android 12 but only after install the magisk Delta via Terminal emulator.apk

**Comment by VisionR1** [comment]
> > > appending everything in `/system/etc/init/magisk.rc` to `/system/etc/init/hw/init.rc`
> > 
> > 
> > Copy content of /system/etc/init/magisk.rc and append to /system/etc/init/hw/init.rc
> 
> Even on my nox android 9 which is installed successfuly dont have magisk.rc in /system/etc/init/magisk.rc

I found a magisk.rc in new version 7.0.5.8 android 12 but only after install the magisk Delta via Terminal emulator.apk

**Comment by VisionR1** [comment]
> 

I try it, copy and paste from the magisk.rc to init.rc but not work, i try and some modifications but still not working, maybe I'm doing something wrong

**Comment by VisionR1** [comment]
> > > > > appending everything in `/system/etc/init/magisk.rc` to `/system/etc/init/hw/init.rc`
> > > > 
> > > > 
> > > > Copy content of /system/etc/init/magisk.rc and append to /system/etc/init/hw/init.rc
> > > 
> > > 
> > > Even on my nox android 9 which is installed successfuly dont have magisk.rc in /system/etc/init/magisk.rc
> > 
> > 
> > I found a magisk.rc in new version 7.0.5.8 android 12 but only after install the magisk Delta via Magisk.Terminal.Emulator.apk
> > ![Nox Player Android 12](https://user-images.githubusercontent.com/25982450/258204482-0ca6fd50-68d0-4e58-8b50-76d63ed275a4.png)
> 
> can u test the steps he metnioned and see if it works as well,

I try it, copy and paste from the magisk.rc to init.rc but not work, i try and some modifications but still not working, maybe I'm doing something wrong

**Comment by VisionR1** [comment]
> > 
> 
> well i found one of the versions of magisk terminal i saved, and i bricked the whole system when i tried to test moving it to Hw/init.rc Lol

Yeah, i try myself some modifications and bricked and install again many times lol

**Comment by VisionR1** [comment]
> > > > 
> > > 
> > > 
> > > well i found one of the versions of magisk terminal i saved, and i bricked the whole system when i tried to test moving it to Hw/init.rc Lol
> > 
> > 
> > Yeah, i try myself some modifications and bricked and install again many times lol
> 
> Im sure it is nox system disk protection/ read only and is up to Husky to try to fix/find a way to unlock the disk properly or for nox to make changes which is highly unlikely as well so prob will not have such feature for a long time

Yeah, unfortunately, now just must we waiting

**Comment by VisionR1** [comment]
> > > > 
> > > 
> > > 
> > > well i found one of the versions of magisk terminal i saved, and i bricked the whole system when i tried to test moving it to Hw/init.rc Lol
> > 
> > 
> > Yeah, i try myself some modifications and bricked and install again many times lol
> 
> Im sure it is nox system disk protection/ read only and is up to Husky to try to fix/find a way to unlock the disk properly or for nox to make changes which is highly unlikely as well so prob will not have such feature for a long time

Yeah, unfortunately, now just must we waiting. The only fix i managed to make is enable the root in nox, after create a /sbin folder manually and after install the Magisk Delta from itself and restart and after when the emulator open, grab the magisk.rc has create by the previous installation with Magisk terminal emulator.apk, disable nox root and restart, and the root works without the root button in nox, but still not work the Zygisk

**Comment by VisionR1** [comment]
> > > > > appending everything in `/system/etc/init/magisk.rc` to `/system/etc/init/hw/init.rc`
> > > > 
> > > > 
> > > > Copy content of /system/etc/init/magisk.rc and append to /system/etc/init/hw/init.rc
> > > 
> > > 
> > > Even on my nox android 9 which is installed successfuly dont have magisk.rc in /system/etc/init/magisk.rc
> > 
> > 
> > I found a magisk.rc in new version 7.0.5.8 android 12 but only after install the magisk Delta via Magisk.Terminal.Emulator.apk
> > ![Nox Player Android 12](https://user-images.githubusercontent.com/25982450/258204482-0ca6fd50-68d0-4e58-8b50-76d63ed275a4.png)
> 
> Can u send the official terminal emulator apk github link, also how did install it which options in the magisk terminal CLI?

The link is down i think, here the last version v17.7(10049) and the options is install in the system and the x option choose magisk delta from a apk file, something like that i think

[v17.7(10049).zip](https://github.com/HuskyDG/magisk-files/files/12281275/v17.7.10049.zip)

**Comment by VisionR1** [comment]
> It seems that the installer does not create `/sbin` folder. Try to create `/sbin` folder?

https://github.com/HuskyDG/magisk-files/issues/119#issuecomment-1666140676

**Comment by VisionR1** [comment]
> It seems that the installer does not create `/sbin` folder. Try to create `/sbin` folder?

@HuskyDG Yeah, i mention this in this comment here (https://github.com/HuskyDG/magisk-files/issues/119#issuecomment-1666140676), before 4 days, still zygisk not work neither riru, i try to merge and modified some files from nox player android 9 and the bluestacks android 11 to nox player android 12, i try modified the bootanim.rc and magisk.rc and the init.rc without success

and here some logs
[magisk_log_2023-08-06T19.10.41.log](https://github.com/HuskyDG/magisk-files/files/12283253/magisk_log_2023-08-06T19.10.41.log)
[magisk_log_2023-08-07T23.48.40.log](https://github.com/HuskyDG/magisk-files/files/12285128/magisk_log_2023-08-07T23.48.40.log)
[magisk_log_2023-08-08T15.49.36.log](https://github.com/HuskyDG/magisk-files/files/12292634/magisk_log_2023-08-08T15.49.36.log)
[magisk_log_2023-08-08T17.35.44.log](https://github.com/HuskyDG/magisk-files/files/12292638/magisk_log_2023-08-08T17.35.44.log)

![Nox Player Android 12 Zygisk](https://github.com/HuskyDG/magisk-files/assets/25982450/74a254d2-eb82-4e2e-8ee7-57d3c21ce861)
![Nox Player Android 12 Riru](https://github.com/HuskyDG/magisk-files/assets/25982450/695ed661-1220-4d50-9715-d872d7c5c159)

**Comment by VisionR1** [comment]
> > It seems that the installer does not create `/sbin` folder. Try to create `/sbin` folder?
> 
> @HuskyDG Yeah, i mention this in this comment here, if you can check it ([#119 (comment)](https://github.com/HuskyDG/magisk-files/issues/119#issuecomment-1666140676)), before 4 days, still zygisk not work neither riru, i try to merge and modified some files from nox player android 9 and the bluestacks android 11 to nox player android 12, i try modified the bootanim.rc and magisk.rc and the init.rc without success
> 
> and here some logs [magisk_log_2023-08-06T19.10.41.log](https://github.com/HuskyDG/magisk-files/files/12283253/magisk_log_2023-08-06T19.10.41.log) [magisk_log_2023-08-07T23.48.40.log](https://github.com/HuskyDG/magisk-files/files/12285128/magisk_log_2023-08-07T23.48.40.log) [magisk_log_2023-08-08T15.49.36.log](https://github.com/HuskyDG/magisk-files/files/12292634/magisk_log_2023-08-08T15.49.36.log) [magisk_log_2023-08-08T17.35.44.log](https://github.com/HuskyDG/magisk-files/files/12292638/magisk_log_2023-08-08T17.35.44.log)
> 
> ![Nox Player Android 12 Zygisk](https://user-images.githubusercontent.com/25982450/258923463-74a254d2-eb82-4e2e-8ee7-57d3c21ce861.png) ![Nox Player Android 12 Riru](https://user-images.githubusercontent.com/25982450/259137836-695ed661-1220-4d50-9715-d872d7c5c159.png)

**Comment by VisionR1** [comment]
I try it and the new version cd3e7684 (26105) testing phase 4, download from telegram, in nox player android 12 without success neither working cant intall, and the strange is not working neither nox player android 9 and bluestacks android 11 before works the last 2 with fdac22ba-delta(25210

**Comment by VisionR1** [comment]
@HuskyDG The Magisk Delta install from itself fixed in all emulators with latest Canary 07c25dab-delta(26105), but still the nox player android 12 not work Zygisk or riru

**Comment by VisionR1** [comment]
> > the nox player android 12 not work Zygisk or riru
> 
> Can't fix.

Can use something like this for fix?

![Zygisk](https://github.com/HuskyDG/magisk-files/assets/25982450/e6ca542a-7afb-4fe9-a176-cc4f211804ef)

**Comment by iPurya** [comment]
> > > > > the nox player android 12 not work Zygisk or riru
> > > > 
> > > > 
> > > > Can't fix.
> > > 
> > > 
> > > Can use something like this for fix? or modify some file in nox to work?
> > > ![Zygisk](https://user-images.githubusercontent.com/25982450/260438060-e6ca542a-7afb-4fe9-a176-cc4f211804ef.png)
> > 
> > 
> > NoxPlay issue does not like this
> 
> Oh understand, thanks for the try to fix it, i try modify some file inside in the nox player android 12 including in the init/hw folder the init.zygote64_32.rc without success the emulator stuck at boot

Did you find a solution ?

**Comment by VisionR1** [comment]
NoxPlayer A12 finally have zygisk

https://github.com/HuskyDG/magisk-files/issues/199
